fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -451,16 +456,30 @@ private CompletableFuture<Void> setIndexingTypeOrThrow(FDBRecordStore store, boo
                                         savedStamp, continuedBuild));
                     }
                     // fall down to exception
-                    throw newPartlyBuildException(continuedBuild, savedStamp, indexingTypeStamp, index);
+                    throw newPartlyBuiltException(continuedBuild, savedStamp, indexingTypeStamp, index);
                 });
     }
 
     private boolean shouldAllowTakeoverContinue(IndexBuildProto.IndexBuildIndexingStamp newStamp, IndexBuildProto.IndexBuildIndexingStamp savedStamp) {
-        return policy.shouldAllowTakeoverContinue() &&
+        return
+            areSimilar(newStamp, savedStamp)
+            ?
+            (!isTypeStampBlocked(savedStamp) || policy.shouldAllowUnblock(savedStamp.getBlockID()))
+            :
+            policy.shouldAllowTakeoverContinue() &&
                (newStamp.getMethod() == IndexBuildProto.IndexBuildIndexingStamp.Method.BY_RECORDS &&
                 savedStamp.getMethod() == IndexBuildProto.IndexBuildIndexingStamp.Method.MULTI_TARGET_BY_RECORDS) ||
                (newStamp.getMethod() == IndexBuildProto.IndexBuildIndexingStamp.Method.BY_RECORDS &&
-               savedStamp.getMethod() == IndexBuildProto.IndexBuildIndexingStamp.Method.MUTUAL_BY_RECORDS);
+                savedStamp.getMethod() == IndexBuildProto.IndexBuildIndexingStamp.Method.MUTUAL_BY_RECORDS);
+    }
+
+    private static boolean areSimilar(IndexBuildProto.IndexBuildIndexingStamp newStamp, IndexBuildProto.IndexBuildIndexingStamp savedStamp) {
+        return savedStamp.getMethod() == newStamp.getMethod()
+               && savedStamp.getTargetIndexList().size() == newStamp.getTargetIndexList().size()
+               && new HashSet<>(savedStamp.getTargetIndexList()).containsAll(newStamp.getTargetIndexList())
```

Comment:
`targetIndex` is supposed to be sorted alphabetically, so you should just be able to do 
```suggestion
               && savedStamp.getTargetIndexList().equals(newStamp.getTargetIndexList())
```

Or are you trying to avoid making that sorting a requirement, and have this code handle the possibility that some code puts it in unsorted.

Reply:
I admit that I hadn't thought about it. But since the code runs rarely, maybe it's better pay the performance penalty and assume nothing about the sorting.